src/astro/sql/operators/upstream_tasks.py

```diff
@@ -0,0 +1,12 @@
+from airflow.models.xcom_arg import XComArg
+
+
+class UpstreamTaskMixin:
+    def __init__(self, **kwargs):
+        upstream_tasks = kwargs.pop("upstream_tasks", [])
+
+        super().__init__(**kwargs)
+
+        for task in upstream_tasks:
+            if isinstance(task, XComArg):
```

Comment:
What if the task is not an instance of `XComArg`, do we need to do any additional steps to set the dependency?

Reply:
@tatiana I've added a condition for if it's a BaseOperator, if the task isn't one of those two then there's nothing for us to upstream.